models/image.py

```diff
@@ -877,23 +674,35 @@ def from_images(cls, images):
         output.exp_time = sum([image.exp_time for image in images])
 
         # start MJD and end MJD
-        output.mjd = min([image.mjd for image in images])
-        output.end_mjd = max([image.end_mjd for image in images])
+        output.mjd = images[0].mjd  # assume sorted by start of exposures
+        output.end_mjd = max([image.end_mjd for image in images])  # exposure ends are not necessarily sorted
 
         # TODO: what about the header? should we combine them somehow?
         output.header = images[0].header
         output.raw_header = images[0].raw_header
 
-        output.source_images = images
         base_type = images[0].type
         if not base_type.startswith('Com'):
             output.type = 'Com' + base_type
 
+        output.upstream_images = images
+
+        # mark the first image as the reference (this can be overriden later when actually warping into one image)
+        output.ref_image_id = output.upstream_images[0].id
```

Comment:
See comments above re: ref_image_id and new_image_id.

What I would do here is add an optional argument "coadd_coord_ref" or some such that stays which image is the coordinate reference.  Then, this code would add entires into the image assocation table with assocation_type of coadd_member.  If coadd_coord_ref is not None, then that one image will have its assocation type be coadd_member_and_coord_ref (or whatever).

Another note: because you have a specific function for making a combined image from a new and ref, I'd put in the docstring that this is not for constructing the data strutures assocated with a subtraction.  It's really only in the case where all of the images are (mostly) filling the same role.  (Can you think of anything other than a coadd where we'd call this?  If we median the images, I'd still call that a "coadd".)

Reply:
I will update the docstring. This should be used for coadds but this also could include a master bias or flat combination.